test/test.ml

```diff
@@ -68,8 +70,8 @@ let test_ring () =
     Logs.(set_level (Some Debug));
     Logs_reporter.(create ~ch:w ~ring_size:2 ~console_threshold clock |> run) @@ fun () ->
     test_console r;
-    Lwt.fail (Failure "Oops")
-  end with Failure "Oops" ->
+    Lwt.fail (Test_failure "Oops")
```

Comment:
maybe remove the string argument completely? e.g. `exception Test_failure` and `Lwt.fail Test_failure`